src/Document/Gateway.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+namespace Conticket\Document;
+
+use Doctrine\ODM\MongoDB\Mapping\Annotations as ODM;
+
+/** @ODM\EmbeddedDocument */
+class Gateway
```

Comment:
If you do not want to make this `final` and opened to extension, can mark this as `abstract` btw